src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -703,15 +703,22 @@ public virtual string UserName
         }
 
         /// <summary>
-        /// SSH Key Path
+        /// SSH Key File Path
         /// </summary>
         [Parameter(ParameterSetName = PSRemotingBaseCmdlet.SSHHostParameterSet)]
         [ValidateNotNullOrEmpty()]
-        public virtual string KeyPath
+        public virtual string KeyFilePath
         {
-            get;
-            set;
+            get { return _keyFilePath; }
+
+            set
+            {
+                // Resolve the key file path.
+                PathResolver resolver = new PathResolver();
+                _keyFilePath = resolver.ResolveProviderAndPath(value, true, this, false, RemotingErrorIdStrings.FilePathNotFromFileSystemProvider);
```

Comment:
Should this be in the setter?  We don't typically validate path arguments this way, it's usually done in BeginProcessing.

Also, it looks like PathResolver should be turned into a static class - it has no data members. You could then avoid the memory allocation for the instance here or wherever the api is used (I see just one other use).